pgx-examples/aggregate/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use pgx::*;
+use std::{
+    str::FromStr,
+    ffi::CStr,
+};
+
+pg_module_magic!();
+
+
+use serde::{Deserialize, Serialize};
+
+
+#[derive(Copy, Clone, PostgresType)]
+#[pgvarlena_inoutfuncs]
+pub struct IntegerAvgState {
+    sum: i32,
+    n: i32,
+}
+
+impl PgVarlenaInOutFuncs for IntegerAvgState {
+    fn input(input: &CStr) -> PgVarlena<Self> {
+        let mut result = PgVarlena::<Self>::new();
+
+        let mut split = input.to_bytes().split(|b| *b == b',');
+        let sum = split.next().map(|value| 
+            i32::from_str(unsafe { std::str::from_utf8_unchecked(value) }).expect("invalid i32")
```

Comment:
In the error message, should we match the Postgres type name?

Reply:
I'm not sure this is important, it's an fairly small example. :)